lang/src/org/partiql/lang/planner/GlobalBindings.kt

```diff
@@ -0,0 +1,50 @@
+package org.partiql.lang.planner
+
+import org.partiql.lang.eval.BindingCase
+import org.partiql.lang.eval.BindingName
+
+/** Indicates the result of an attempt to resolve a global binding. */
+sealed class ResolutionResult {
+    /**
+     * A success case, indicates the [uniqueId] of the match to the [BindingName] in the global scope.
+     * Typically, this is defined by the storage layer.
+     */
+    data class GlobalVariable(val uniqueId: String) : ResolutionResult()
+
+    /**
+     * A success case, indicates the [index] of the only possible match to the [BindingName] in a local lexical scope.
+     * This is `internal` because [index] is an implementation detail that shouldn't be accessible outside of this
+     * library.
+     */
+    internal data class LocalVariable(val index: Int) : ResolutionResult()
+
+    /** A failure case, indicates that resolution did not match any variable. */
+    object Undefined : ResolutionResult()
+}
+
+fun interface GlobalBindings {
```

Comment:
Maybe rename this as `GlobalVariableToIdBindings`? The current naming might be confused with the global bindings for evaluation.

Reply:
Going with `UniqueIdResolver`